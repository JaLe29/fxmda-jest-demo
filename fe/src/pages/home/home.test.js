import React from 'react';
import {shallow} from 'enzyme';

import Home from './home';

jest.mock('./user', () => () => <div />)

describe('Home', () => {
	it('renders home component', () => {
		const wrapper = shallow(<Home reportLastUserId={jest.fn()} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should call reportLastUserId with correct id', () => {
		const mockReportLastUserId = jest.fn()
		const wrapper = shallow(<Home reportLastUserId={mockReportLastUserId} />);

		wrapper.find('.reportButton').simulate('click')

		expect(mockReportLastUserId).toHaveBeenCalledWith(4)
	});
});