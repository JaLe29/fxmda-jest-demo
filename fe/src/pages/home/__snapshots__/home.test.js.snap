// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders home component 1`] = `
<div>
  <h1>
    HOME PAGE
  </h1>
  <p>
    xx
  </p>
  <div>
    <button
      onClick={[Function]}
    >
      Add random
    </button>
    <button
      className="reportButton"
      onClick={[Function]}
    >
      Report last user id
    </button>
  </div>
  <hr />
  <Component
    age={27}
    id={1}
    key="1"
    name="Jakub"
  />
  <Component
    age={33}
    id={2}
    key="2"
    name="Pepa"
  />
  <Component
    age={21}
    id={3}
    key="3"
    name="Petr"
  />
  <Component
    age={20}
    id={4}
    key="4"
    name="Honza"
  />
</div>
`;
